### ISSUE TITLE
[BUG][JAVA] SpringCodegen (<library>spring-boot</library>) and BeanValidation for type: string and format: email does not work on version 6.0.1

### ISSUE THREAD

**Issue by JiyangWuUSB** [OpenAPITools/openapi-generator#13379]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

We have a property with type: string and format: email and we use Bean Validation for spring-boot.

      properties:
        email:
          type: string
          minLength: 1
          maxLength: 6
          format: email

Using such validations:
                                   `<useBeanValidation>true</useBeanValidation>`
                                    `<performBeanValidation>true</performBeanValidation>`

When we call this:
`mvn -Dmaven.wagon.http.ssl.insecure=true -Dmaven.wagon.http.ssl.allowall=true clean install`

We got compile error:
`Email is ambiguous
[ERROR]   both class org.hibernate.validator.constraints.Email in org.hibernate.validator.constraints and class javax.validation.constraints.Email in javax.validation.constraints match`

while the expected output is: builds and runs successfully

A previous issue addressed the same error [[BUG][JAVA] SpringCodegen (<library>spring-boot</library>) and BeanValidation for type: string and format: email does not work #4786](https://github.com/OpenAPITools/openapi-generator/issues/4786)

##### openapi-generator version

6.0.1

##### Steps to reproduce
`mvn -Dmaven.wagon.http.ssl.insecure=true -Dmaven.wagon.http.ssl.allowall=true clean install`

##### Related issues/PRs

https://github.com/OpenAPITools/openapi-generator/issues/4786

##### Suggest a fix

Using version 5.3.1 and 5.4.0 does not cause this error, but 6.0.1 creates this error.
If we set `<performBeanValidation>` from `true` to `false`, the error is gone.
If we comment out the line ` format: email`, the error is gone.
But we do now know what are the potential downsides of the two methods mentioned above

**Comment by radekzukowski** [comment]
Any update on this?

**Comment by JiyangWuUSB** [comment]
> Any update on this?

Not much, maybe we need to try v6.1.0 or v6.2.0 first

**Comment by kidhack83** [comment]
In v6.2.1 the error continues:

```
[INFO] --- openapi-generator-maven-plugin:6.2.1:generate (generate-objects) @ ... ---
[ERROR] /C:/.../GuestDto.java:[130,28] reference to Email is ambiguous
  both class org.hibernate.validator.constraints.Email in org.hibernate.validator.constraints and class javax.validation.constraints.Email in javax.validation.constraints match
```

**Comment by Mintas** [comment]
@kidhack83  @JiyangWuUSB  @radekzukowski coming soon

**Comment by Mintas** [comment]
not yet, partially my fault, coming back to this bugfix soon

**Comment by vivekvara** [comment]
Can someone please merge PR if fixed?

**Comment by fabiofranco85** [comment]
are there any workarounds for this issue?

**Comment by Mintas** [comment]
@fabiofranco85  Hello!
yes there is =)
since noone merged this PR for 3 months, You have to do the following:
1) copy contents of beanValidationCore.mustache from https://github.com/OpenAPITools/openapi-generator/pull/13962/files#diff-2b6ec1643b8bc9fd8f22d34288fc2f73f153f441944328bcf5a2fd7468d9e8ab
2) add it to your build folder https://openapi-generator.tech/docs/templating/
3) verify it works by building / generating open-api specs
4) have fun!

**Comment by deepakthankachan** [comment]
Has this fixed been merged? Still getting the same error on v7.2.0 :(
